lib/src/metrics/weight-of-class/weight_of_class_metric.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:code_checker/checker.dart';
+
+import '../../utils/declaration_utils.dart';
+
+class WeightOfClassMetric {
+  double compute(
+    ScopedClassDeclaration component,
+    Iterable<ScopedFunctionDeclaration> functions,
+  ) {
+    final totalPublicMethods = componentFunctions(component, functions)
+        .where((function) => !Identifier.isPrivateName(
+              getDeclarationHumanReadableName(function.declaration),
+            ))
+        .toList();
+
+    final functionalMethods = totalPublicMethods.where((function) {
+      if (function.type == FunctionType.constructor) {
+        return false;
+      }
+
+      final declaration = function.declaration;
+      if (declaration is MethodDeclaration) {
+        return !declaration.isGetter && !declaration.isGetter;
```

Comment:
Looks like there should be `setter` check

Reply:
fix